internal/watcher/process_watcher.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package watcher
+
+import (
+	"context"
+
+	"github.com/nginx/agent/v3/internal/model"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+type (
+	ProcessWatcher struct{}
```

Comment:
Could we maybe call this an operator (it's stateless and is called by the instance watcher)? Don't want to muddy the water on the difference between an operator and a watcher this soon in the refactor 😄

Reply:
yeah good point. Let me update that.